causalpy/experiments/interrupted_time_series.py

```diff
@@ -25,16 +26,279 @@
 from patsy import build_design_matrices, dmatrices
 from sklearn.base import RegressorMixin
 
-from causalpy.custom_exceptions import BadIndexException
+from causalpy.custom_exceptions import BadIndexException, ModelException
+from causalpy.experiments.base import BaseExperiment
 from causalpy.plot_utils import get_hdi_to_df, plot_xY
 from causalpy.pymc_models import PyMCModel
 from causalpy.utils import round_num
 
-from .base import BaseExperiment
-
 LEGEND_FONT_SIZE = 12
 
 
+class TreatmentTimeHandler(ABC):
```

Comment:
Should abstract classes sit in the ITS class i.e. will it be used for any other time-based causal inference methods?